yang/frr-bgp.yang

```diff
@@ -0,0 +1,1248 @@
+        description
+          "Global configuration for the BGP router.";
+        leaf local-as {
+          type inet:as-number;
+          mandatory true;
+          description
+            "Local autonomous system number of the router.  Uses
+             the 32-bit as-number type from the model in RFC 6991.";
+        }
+
+        leaf instance-type {
+          type frr-bt:bgp-instance-type;
+          mandatory true;
+          description
+            "BGP instance type";
+        }
+
+        uses frr-rt:router-id;
+
+        container confederation {
+          description
+            "Configuration options specifying parameters when the
+             local router is within an autonomous system which is
+             part of a BGP confederation.";
+          leaf identifier {
+            type inet:as-number;
+            description
+              "Confederation identifier for the autonomous system.";
+          }
+
+          leaf-list member-as {
+            type inet:as-number;
+            description
+              "Remote autonomous systems that are to be treated
+               as part of the local confederation.";
+          }
+        }
+
+        uses med-config;
+
+        uses route-reflector-config;
+
+        uses route-selection-options;
+
+        uses global-neighbor-config;
+
+        container graceful-restart {
+          description
+            "Parameters relating the graceful restart mechanism for
+             BGP.";
+          uses graceful-restart-config;
+        }
+
+        uses global-update-group-config;
+
+        uses global-config-timers;
+
+        uses global-bgp-config;
+
+        uses global-static-route-config;
+
+        uses global-graceful-shutdown;
+
+        uses global-bmp-config;
+
+        container afi-safis {
+          description
+            "List of address-families associated with the BGP
+             instance.";
+          list afi-safi {
+            key "afi-safi-name";
+            description
+              "AFI, SAFI configuration available for the
+               neighbour or group.";
+            uses mp-afi-safi-config;
+
+            uses mp-all-afi-safi-list-contents;
+          }
+        }
+      }
+
+      container neighbors {
+        description
+          "Configuration for BGP neighbors.";
+        list neighbor {
+          key "remote-address";
+          description
+            "List of BGP neighbors configured on the local system,
+             uniquely identified by remote IPv[46] address.";
+          leaf remote-address {
+            type inet:ip-address;
+            description
+              "The remote IP address of this entry's BGP peer.";
+          }
+
+          leaf local-interface {
+            type frr-interface:interface-ref {
+              require-instance false;
+            }
+            description
+              "The local interface for the TCP connection between
+               the BGP peers.";
```

Comment:
Confused about what this means. Is this operational state? If not, what does it mean to configure the "local interface"?

Reply:
This is getting mapped to

DEFUN (neighbor_port,
       neighbor_port_cmd,
       "neighbor <A.B.C.D|X:X::X:X> port (0-65535)",
       NEIGHBOR_STR
       NEIGHBOR_ADDR_STR
       "Neighbor's BGP port\n"
       "TCP port number\n")